documentation/projects/proposals/dark_mode/20240325-implementation_plan_dark_mode.md

```diff
@@ -0,0 +1,224 @@
+override the default behavior or add dark-mode specific styles beyond the core
+palette swap.
+
+Here are pretend, simplified examples of this setup.
+
+In our primary CSS file, we setup two lists of CSS variables with different
+values when the `.dark-mode` class is present:
+
+```css
+:root {
+  --color-foreground: black;
+  --color-background: white;
+}
+
+.dark-mode {
+  --color-foreground: white;
+  --color-background: black;
+}
+```
+
+In our Tailwind config, we reference these variables:
+
+```js
+const config = {
+  theme: {
+    colors: {
+      foreground: "var(--color-foreground)",
+      background: "var(--color-background)",
+    },
+  },
+}
+```
+
+In a component, we use the Tailwind classes:
+
+```vue
+<template>
+  <!-- This will be black in light mode and white in dark mode! -->
+  <p class="text-foreground">Hello World</p>
+</template>
+```
+
+Finally, if we needed an "escape hatch" to make sure this component was _always_
+black, _regardless_ of dark mode:
+
+```vue
+<template>
+  <p class="text-foreground dark:text-background">Hello World</p>
+</template>
+```
+
+## Expected Outcomes
+
+<!-- List any succinct expected products from this implementation plan. -->
+
+- Users writing components will have a "dark mode compatible by default"
+  experience. By default, they will not need to think much about dark mode.
+- Color names in the codebase will be replaced with semantic names.
+- Frontend developers will have easy tools to visually test components in light
+  and dark mode.
+
+## Rejected alternate approaches
+
+- Using the `dark:` modifier _exclusively_ for dark mode styling. This is more
+  explicit, but much, much more verbose, and would require extensive edits to
+  every single component we have written. Instead of writing `bg-background`,
+  for example, we would have to write `bg-white dark:bg-black` all throughout
+  the codebase.
+
+## Step-by-step plan
+
+<!--
+List the ordered steps of the plan in the form of imperative-tone issue titles.
+
+The goal of this section is to give a high-level view of the order of implementation any relationships like
+blockages or other dependencies that exist between steps of the plan. Link each step to the step description
+in the following section.
+
+If special deployments are required between steps, explicitly note them here. Additionally, highlight key
+milestones like when a feature flag could be made available in a particular environment.
+-->
+
+The following plan requires approved designs and semantic color names.
+
+1. Parallel Work Stream "A": Implement the new color palette
+   1. Create a `FORCE_DARK_MODE` feature flag which is disabled by default and
+      available in every environment. Add logic to add a `dark-mode` class to
+      the root HTML tag when this flag is enabled and a `light-mode` class when
+      disabled.
+   2. Rename all colors in the tailwind config and the frontend components to
```

Comment:
This step also doesn't seem to be dependent on anything else and could be done in parallel.